arcgis-java-toolkit/src/main/java/com/esri/arcgisruntime/toolkit/UtilityNetworkTraceTool.java

```diff
@@ -0,0 +1,999 @@
+
+  /**
+   * Property that determines what symbol should be used to denote polylines in a geometry trace result.
+   * Default is a blue, dot style line.
+   *
+   * @return the resultLineSymbol property
+   * @since 100.15.0
+   */
+  public SimpleObjectProperty<SimpleLineSymbol> resultLineSymbolProperty() { return resultLineSymbolProperty; }
+
+  /**
+   * Gets the symbol used to denote polylines in a geometry trace result.
+   *
+   * @return the symbol used for polylines
+   * @since 100.15.0
+   */
+  public SimpleLineSymbol getResultLineSymbol() { return resultLineSymbolProperty.get(); }
+
+  /**
+   * Sets the symbol used to denote polylines in a geometry trace result.
+   *
+   * @param simpleLineSymbol the symbol to use for polylines
+   * @since 100.15.0
+   */
+  public void setResultLineSymbol(SimpleLineSymbol simpleLineSymbol) { resultLineSymbolProperty.set(simpleLineSymbol); }
+
+  /**
+   * Property that determines what symbol should be used to denote polygons in a geometry trace result.
+   * Default is a blue, dot style line.
+   *
+   * @return the resultFillSymbol property
+   * @since 100.15.0
+   */
+  public SimpleObjectProperty<SimpleFillSymbol> resultFillSymbolProperty() { return resultFillSymbolProperty; }
+
+  /**
+   * Gets the symbol used to denote polygons in a geometry trace result.
+   *
+   * @return the symbol used for polygons
+   * @since 100.15.0
+   */
+  public SimpleFillSymbol getResultFillSymbol() {return resultFillSymbolProperty.get(); }
+
+  /**
+   * Sets the symbol used to denote polygons in a geometry trace result.
+   *
+   * @param simpleFillSymbol the symbol to use for polygons
+   * @since 100.15.0
+   */
+  public void setResultFillSymbol(SimpleFillSymbol simpleFillSymbol) { resultFillSymbolProperty.set(simpleFillSymbol); }
+
+  /**
+   * Gets any UtilityNetworks from the ArcGIS Map attached to the MapView. The ArcGIS Map must be loaded in order to
+   * access the Utility Network data. In addition, the Utility Networks are loaded before setting to the relevant
+   * properties. Messages are logged if any of these requirements are not met.
+   *
+   * @since 100.15.0
+   */
+  private void setupUtilityNetworks() {
+    isMapAndUtilityNetworkLoadingInProgressProperty.set(true);
+    var mapView = mapViewProperty.get();
+    // check if there is map
+    if (mapView.getMap() != null) {
+      ArcGISMap map = mapViewProperty.get().getMap();
+      map.addDoneLoadingListener(() -> {
+        if (map.getLoadStatus() == LoadStatus.LOADED) {
+          // check if the map has utility networks
+          if (!map.getUtilityNetworks().isEmpty()) {
+            ObservableList<UtilityNetwork> utilityNetworksFromMap = FXCollections.observableArrayList();
+            // create completable futures to wait for all utility networks to finish loading before
+            // setting the data
+            CompletableFuture<?>[] futures = new CompletableFuture<?>[map.getUtilityNetworks().size()];
```

Comment:
Do the futures actually need to be indexed? If not using a list here would be simpler.

Reply:
No they didn't, I've refactored.